lib/FrontendTool/FrontendTool.cpp

```diff
@@ -237,12 +237,154 @@ template <> struct ObjectTraits<LoadedModuleTraceFormat> {
 }
 }
 
+namespace {
+
+// The most interesting/tricky thing here is _which_ paths get recorded. It
+// depends on how the module was synthesized. The key points are:
+//
+// 1. Paths in the cache are not recorded - Precondition: the corresponding
+//    path to the swiftinterface must already be present as a key in
+//    pathToModuleDecl.
+// 2. swiftmodules next to a swiftinterface are saved if they are up-to-date.
```

Comment:
This comment is missing a bunch of context for a reader just coming upon it. What paths are you talking about? What cache?

Reply:
Good catch. How about I rephrase it to

> Paths pointing to swiftmodules on the module cache path or in the prebuilt cache are not recorded

Does that sound reasonable? More context needed?

Reply:
I've never heard "on" used to refer to directories, only filesystems/mounts/disks.

Reply:
Changed it to "Paths to swiftmodules in the module cache or in the prebuilt cache"

Reply:
I still think this comment doesn't make sense for someone just reading along. To put it another way, if you made this a proper doc comment, it's missing the very first part that describes what the function does.

Reply:
I've added a line of context on top.